spring-cloud-dataflow-server/docker-compose-postgres.yml

```diff
@@ -22,17 +22,17 @@ services:
 
   dataflow-server:
     environment:
-      - SPRING_DATASOURCE_URL=jdbc:postgresql://postgres:5432/dataflow
-      - SPRING_DATASOURCE_USERNAME=root
-      - SPRING_DATASOURCE_PASSWORD=rootpw
-      - SPRING_DATASOURCE_DRIVER_CLASS_NAME=org.postgresql.Driver
+      - spring.datasource.url=jdbc:postgresql://postgres:5432/dataflow
```

Comment:
@tzolov The way I understood was that, the properties under `environment` would always be ENV properties irrespective of the case. Can you help me understand how this works this way?

Reply:
@ilayaperumalg , when you deploy an app you can set apps's own `--spring.datasource.url` that would override the one coming form the Skipper. But if skipper uses env variables (e.g. `SPRING_DATASOURCE_URL`) then apparently children's property is ignored in favor of the inherited env variable.  I guess in boot the env. vars has precedence over properties. 
The above behaviors observed by the local deployers used by the docker-compose configurations.

Reply:
That's a bit weird as it's still coming from a boot's property resolver which resolves from environment and thus format in it should not make any difference.

Reply:
@jvalkeal Do you think this PR needs to change? Maybe there are some caveats we miss? What do you think @tzolov ?

Reply:
Lets say that environment variables have `SPRING_DATASOURCE_URL` and `spring.datasource.url` a `XXX_XXX_XXX takes precedence. Then this may be a symptom of how local deployer is setting properties into environment variables using dot style so it would override that but not the normal underscore type. Running apps in a same container is not so supported scenario being testing. Maybe local deployer could be setup not to inherit those database values automatically?